src/app/qgsvectorlayerloadstyledialog.cpp

```diff
@@ -141,24 +142,27 @@ void QgsVectorLayerLoadStyleDialog::initializeLists( const QStringList &ids, con
   mRelatedTable->setHorizontalHeaderLabels( relatedTableHeader.split( ';' ) );
   mOthersTable->setHorizontalHeaderLabels( othersTableHeader.split( ';' ) );
   mRelatedTable->setRowCount( sectionLimit );
-  mOthersTable->setRowCount( ids.count() - sectionLimit );
+  mOthersTable->setRowCount( sectionLimit >= 0 ? ( ids.count() - sectionLimit ) : 0 );
   mRelatedTable->setDisabled( relatedTableNOfCols == 1 );
   mOthersTable->setDisabled( othersTableNOfCols == 1 );
 
-  for ( int i = 0; i < sectionLimit; i++ )
+  if ( sectionLimit > 0 )
```

Comment:
this is redundant, if sectionLimit = 0, nothing will be done by the for loop
do you prefer to keep it?

Reply:
Look at the second loop, that is the one that crashed when sectionlimit was -1

Reply:
right, sorry!

Reply:
np, thanks for reviewing!